src/decision/src/GpsDecision.cpp

```diff
@@ -36,29 +30,48 @@ GpsDecision::GpsDecision(int argc, char **argv, std::string node_name) {
     // based on our distance and heading to the destination GPS waypoint
     double linear_heading_factor, linear_distance_factor,
             angular_heading_factor, angular_distance_factor;
-    double linear_cap, angular_cap;
     SB_getParam(private_nh, "linear_heading_factor", linear_heading_factor, 1.0);
     SB_getParam(private_nh, "linear_distance_factor", linear_distance_factor, 1.0);
     SB_getParam(private_nh, "angular_heading_factor", angular_heading_factor, 1.0);
     SB_getParam(private_nh, "angular_distance_factor", angular_distance_factor, 1.0);
     mover.setFactors(linear_distance_factor, linear_heading_factor,
                     angular_distance_factor, angular_heading_factor);
+
+    double linear_cap, angular_cap;
     SB_getParam(private_nh, "max_linear_speed", linear_cap, 1.0);
     SB_getParam(private_nh, "max_angular_speed", angular_cap, 1.0);
     mover.setMaxSpeeds(linear_cap, angular_cap);
-}
-
-void GpsDecision::currentLocationCallback(const geometry_msgs::Point::ConstPtr &current_location) {
-    this->current_location = *current_location;
-}
 
-void GpsDecision::imuCallback(const std_msgs::Float32::ConstPtr &heading) {
-    this->current_heading = heading->data;
+    SB_getParam(private_nh, "base_frame", base_frame, (std::string)"base_link");
+    SB_getParam(private_nh, "global_frame", global_frame, (std::string)"odom_combined");
 }
 
-void GpsDecision::waypointCallback(const geometry_msgs::Point::ConstPtr &waypoint) {
-    // Create a new twist message and publish it
-    geometry_msgs::Twist twist = mover.createTwistMessage(current_location, current_heading, *waypoint);
-    twist_publisher.publish(twist);
+void GpsDecision::waypointCallback(const geometry_msgs::PointStamped::ConstPtr &waypoint) {
+    // Get the current position and heading of the robot
+    tf2_ros::Buffer tfBuffer;
```

Comment:
Gahh, tried this and remembered why I didn't do it in the first place. We can only create the `TransformListener` **after** we run `ros::init`. I could resolve this by making the class member a pointer, which I could initialise in the constructor, but given that this isn't that performance sensitive, and if we're not getting a brand new TF we shouldn't be moving anyhow, I'm going to leave this as is.